src/domains/point_set.jl

```diff
@@ -25,34 +27,44 @@ julia> PointSet([(rand(),rand()) for i in 1:100])
 ```
 """
 struct PointSet{T,N} <: SpatialDomain{T,N}
-  coords::Matrix{T}
+  coords::Vector{SVector{N,T}} 
 
   function PointSet{T,N}(coords) where {N,T}
     @assert !isempty(coords) "coordinates must be non-empty"
     new(coords)
   end
 end
 
-PointSet(coords::AbstractMatrix{T}) where {T} =
-  PointSet{T,size(coords,1)}(coords)
+function PointSet(coords::AbstractMatrix{T}) where {T}
+  N = size(coords, 1)
+  points = Vector{SVector{N,T}}()
+  for row in eachcol(coords)
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
+end
 
-PointSet(coords::AbstractVector{NTuple{N,T}}) where {N,T} =
-  PointSet([c[i] for i in 1:N, c in coords])
+function PointSet(coords::AbstractVector{NTuple{N,T}}) where {N,T}
+  points = Vector{SVector{N,T}}()
+  for row in coords
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
+end
 
-nelms(ps::PointSet) = size(ps.coords, 2)
+nelms(ps::PointSet) = size(ps.coords,1)
 
 function coordinates!(buff::AbstractVector{T}, ps::PointSet{T,N},
                       location::Int) where {N,T}
   @inbounds for i in 1:N
-    buff[i] = ps.coords[i,location]
+    buff[i] = ps.coords[location][i]
   end
 end
-
 # ------------
 # IO methods
 # ------------
 function Base.show(io::IO, ps::PointSet{T,N}) where {N,T}
-  npts = size(ps.coords, 2)
+  npts = nelms(ps)
```

Comment:
Let's use `length(ps.coords)` here as well just to avoid introducing dependencies between different function implementations.

Reply:
OK